### PR TITLE
[WIP] Utility to move all derivative terms to LHS 

### PR DESCRIPTION
**Pull request by ranjanan**
So far, works on: 
```julia
julia> eq = x ~ D(x) + Constant(0.5)
Equation(x, derivative(x, t) + 0.5)

julia> ModelingToolkit.movediffvars(eq)
Equation(-1 * derivative(x, t) + x, Constant(0.5))

julia> eq = x ~ D(x) * Constant(0.5)
Equation(x, derivative(x, t) * 0.5)

julia> ModelingToolkit.movediffvars(eq)
Equation(-0.5 * derivative(x, t) + x, Constant(0.0))

```

Doesn't work on: 
```julia
eq = x ~ Constant(5.0) * (D(x) + Constant(0.5))
```
Would be nice to have simplification based on distributive rule (https://github.com/JuliaSymbolics/SymbolicUtils.jl/pull/95)

### CODE REVIEW

File: src/utils.jl [hunk 1]
```diff
@@ -79,6 +79,30 @@ function vars!(vars, O)
...
+			end
+		elseif isa(arg, Operation)
+			movediffvars(arg, ops)
+		end
+	end
+end
+function movediffvars(eqn)
+	ops = Vector{Any}()
+	movediffvars(eqn.rhs, ops)
+	Equation(simplify(eqn.lhs + reduce(+, ops)), simplify(eqn.rhs + reduce(+, ops)))
```
Comment (YingboMa) [inline src/utils.jl:103]:
```suggestion
	Equation(simplify(eqn.lhs + sum(ops)), simplify(eqn.rhs + sum(ops)))
```

**Comment by ChrisRackauckas** [comment]
Shouldn't we move all terms to the RHS for consistency?

**Comment by YingboMa** [comment]
Yeah, we need another pass (or to modify this pass) to move non-differential terms to the RHS, too.

**Comment by ranjanan** [comment]
Yes, I'll write another pass to move the non-differential terms to the RHS.

**Comment by ChrisRackauckas** [comment]
This code is old enough that it's better to start from scratch.

**Pull request closed without merging**